MediaGalleryUi/Model/FilterParametersProvider/Fulltext.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryUi\Model\FilterParametersProvider;
+
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\DB\Select;
+use Magento\MediaGalleryUi\Model\SelectModifierInterface;
+
+/**
+ * Apply fulltext filter
+ */
+class Fulltext implements SelectModifierInterface
+{
+    private const TABLE_ALIAS = 'main_table';
+    private const FULLTEXT_CONDITION_TYPE = 'fulltext';
+    private const TABLE_ASSET_KEYWORD = 'media_gallery_keyword';
+
+    /**
+     * @inheritdoc
+     */
+    public function apply(Select $select, SearchCriteriaInterface $searchCriteria): void
+    {
+        $connection = $select->getConnection();
```

Comment:
Better to initialise a connection if `$value` not `null`. Otherwise, it is not used in case of `$value === null`